controllers/gatekeeper_controller.go

```diff
@@ -193,8 +204,13 @@ func (r *GatekeeperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	err := r.Get(ctx, req.NamespacedName, gatekeeper)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
+			if r.cpsCtrlCtx != nil && r.isCPSCtrlRunning {
+				r.StopCPSController()
```

Comment:
So this requests a stop but doesn't actually wait for it to stop. So what happens if I delete the Gatekeeper object and recreate immediately? It could try to start it again while it's still shutting down.

`StopCPSController` probably shouldn't set `r.isCPSCtrlRunning = false` or have some way to know that it actually stopped before returning.

Reply:
um.... should I check `childeCtx == nil`?  is this enough?

Reply:
This should be resolved based on the discussion with the suggested Git diff.